javascript-carousel/main.js

```diff
@@ -0,0 +1,44 @@
+var $image = document.querySelectorAll('.image');
+var $progressIndicators = document.querySelector('.progress-indicators');
+var $circles = document.querySelectorAll('.circle');
+var $leftArrow = document.querySelector('.left');
+var $rightArrow = document.querySelector('.right');
+var position = 0;
+
+var interval = setInterval(rotate, 3000);
+
+$progressIndicators.addEventListener('click', function (e) {
+  var view = e.target.dataset.view;
+  updateImage(view);
+});
+
+$rightArrow.addEventListener('click', rotate);
+
+$leftArrow.addEventListener('click', function () {
+  position--;
+  updateImage(position);
+  if (position === -1) {
+    updateImage('4');
+  }
```

Comment:
You can simplify your logic a bit by establishing the correct position _first_, then actually updating the image.
```js
if (position === 0) {
  position = $image.length - 1;
} else {
  position--;
}
updateImage(position);
```